crates/task-impls/src/quorum_proposal/dependency_handle.rs

```diff
@@ -0,0 +1,274 @@
+            data: proposal,
+            signature,
+            _pd: PhantomData,
+        };
+        debug!(
+            "Sending proposal for view {:?}",
+            proposed_leaf.view_number(),
+        );
+
+        self.consensus
+            .write()
+            .await
+            .update_last_proposed_view(self.view_number)?;
+        async_sleep(Duration::from_millis(self.round_start_delay)).await;
+        broadcast_event(
+            Arc::new(HotShotEvent::QuorumProposalSend(
+                message.clone(),
+                self.public_key.clone(),
+            )),
+            &self.sender,
+        )
+        .await;
+
+        Ok(())
+    }
+}
+impl<TYPES: NodeType> HandleDepOutput for ProposalDependencyHandle<TYPES> {
+    type Output = Vec<Vec<Vec<Arc<HotShotEvent<TYPES>>>>>;
+
+    #[allow(clippy::no_effect_underscore_binding)]
+    async fn handle_dep_result(self, res: Self::Output) {
+        let mut payload_commitment = None;
+        let mut commit_and_metadata: Option<CommitmentAndMetadata<TYPES>> = None;
+        let mut timeout_certificate = None;
+        let mut view_sync_finalize_cert = None;
+        let mut vid_share = None;
+        for event in res.iter().flatten().flatten() {
+            match event.as_ref() {
+                HotShotEvent::QuorumProposalValidated(proposal, _) => {
+                    let proposal_payload_comm = proposal.block_header.payload_commitment();
+                    if let Some(comm) = payload_commitment {
+                        if proposal_payload_comm != comm {
+                            return;
+                        }
+                    } else {
+                        payload_commitment = Some(proposal_payload_comm);
+                    }
+                }
+                HotShotEvent::SendPayloadCommitmentAndMetadata(
+                    payload_commitment,
+                    builder_commitment,
+                    metadata,
+                    view,
+                    fee,
+                ) => {
+                    commit_and_metadata = Some(CommitmentAndMetadata {
+                        commitment: *payload_commitment,
+                        builder_commitment: builder_commitment.clone(),
+                        metadata: metadata.clone(),
+                        fee: fee.clone(),
+                        block_view: *view,
+                    });
+                }
+                HotShotEvent::QcFormed(cert) => match cert {
+                    either::Right(timeout) => {
+                        timeout_certificate = Some(timeout.clone());
+                    }
+                    either::Left(_) => {
+                        // Handled by the HighQcUpdated event.
+                    }
+                },
+                HotShotEvent::ViewSyncFinalizeCertificate2Recv(cert) => {
+                    view_sync_finalize_cert = Some(cert.clone());
+                }
+                HotShotEvent::ProposeNow(_, pdd) => {
```

Comment:
A later PR is going to completely re work this, so you can ignore checking this.